contracts/superswap/RexSuperSwap.sol

```diff
@@ -0,0 +1,143 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+pragma abicoder v2;
+
+import { ISuperToken } from "@superfluid-finance/ethereum-contracts/contracts/interfaces/superfluid/ISuperfluid.sol";
+import "./TransferHelper.sol";
+import "./interfaces/ISwapRouter02.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../ISETHCustom.sol";
+
+import "hardhat/console.sol";
+
+
+contract RexSuperSwap {
+  using SafeERC20 for ERC20;
+  ISwapRouter02 public immutable swapRouter;
+  address public maticAddress;
+
+  event SuperSwapComplete(uint256 amountOut);
+  event ErrorOnSwap(string reason);
+  event ReturnDataEvent(bytes returnData);
+
+  constructor(ISwapRouter02 _swapRouter, address _maticAddress) {
```

Comment:
`_maticAddress` -> `superNativeToken`, just to avoid confusion with other chains.

Reply:
Ah yeah I understand now, variable has been changed!